core/src/main/scala/org/apache/spark/network/netty/NettyBlockTransferService.scala

```diff
@@ -49,18 +49,21 @@ class NettyBlockTransferService(conf: SparkConf, securityManager: SecurityManage
   private[this] var appId: String = _
 
   override def init(blockDataManager: BlockDataManager): Unit = {
-    val (rpcHandler: RpcHandler, bootstrap: Option[TransportClientBootstrap]) = {
-      val nettyRpcHandler = new NettyBlockRpcServer(serializer, blockDataManager)
-      if (!authEnabled) {
-        (nettyRpcHandler, None)
+    val rpcHandler = new NettyBlockRpcServer(serializer, blockDataManager)
+    val (serverBootstrap: Option[TransportServerBootstrap],
```

Comment:
nit: This syntax gets unwieldy with more code in here, I'd advise switching to something like
```scala
var serverBootstrap: Option[TransportServerBootstrap] = None
var clientBootstrap: Option[TransportClientBootstrap] = None
if (authEnabled) {
  serverBootstrap = Some(new SaslServerBootstrap(transportConf, securityManager)
  clientBootstrap = Some(new SaslClientBootstrap(transportConf, conf.getAppId, securityManager,
    securityManager.isSaslEncryptionEnabled()))
}
```

uses the dreaded `var` but I think it may be less syntactically cumbersome.